File: Programming_Basics_with_Python/nested_loop_exercise/04._train_the_trainers.py
number_of_jury = int(input())
final_grade = 0
number_of_presentations = 0
command = input()
while command != "Finish":
    current_presentations_name = command
    number_of_presentations += 1
    current_presentations_grade = 0
    for presentation_grade in range(number_of_jury):
        grade = float(input())
        current_presentations_grade += grade
    presentation_average_grade = current_presentations_grade/number_of_jury
    final_grade += presentation_average_grade
    print(f'{current_presentations_name} - {presentation_average_grade:.2f}.')
    command = input()
final_average_grade = final_grade / number_of_presentations
print(f"Student's final assessment is {final_average_grade:.2f}.")

File: Programing_Basics_with_JS/nestedLoopsExercise/04.TrainTheTrainers.js
function trainTheTrainers(input) {
    let index = 0;
    let numbersOfJury = Number(input[index]);
    index++;
    let command = input[index];
    index++;

    let presentationCount = 0;
    let presentationTotalGrade = 0;

    while(command !== 'Finish') {
        let presentationName = command;
        presentationCount++;
        let courseTotalGrade = 0;

        for(let n = 1; n <= numbersOfJury; n++) {
            let courseGrade = Number(input[index]);
            index++;
            courseTotalGrade += courseGrade;
        }

        let courseAverageGrade = courseTotalGrade / numbersOfJury;
        presentationTotalGrade += courseAverageGrade;
        console.log(`${presentationName} - ${courseAverageGrade.toFixed(2)}.`);

        command = input[index];
        index++;
    }

    let presentationAverageGrade = presentationTotalGrade / presentationCount;
    console.log(`Student's final assessment is ${presentationAverageGrade.toFixed(2)}.`);
}

trainTheTrainers(["2",
"While-Loop",
"6.00",
"5.50",
"For-Loop",
"5.84",
"5.66",
"Finish"])


trainTheTrainers(["3",
"Arrays",
"4.53",
"5.23",
"5.00",
"Lists",
"5.83",
"6.00",
"5.42",
"Finish"])

trainTheTrainers(["2",
"Objects and Classes",
"5.77",
"4.23",
"Dictionaries",
"4.62",
"5.02",
"RegEx",
"2.88",
"3.42",
"Finish"])


